### ISSUE TITLE
Include immediate static module dependencies in the traced result

### ISSUE THREAD

**Issue by prantlf** [amodrojs/amodro-trace#3]
I work on a SPA consisting of roughly 900 modules divided to 4 bundles, all running in the browser.  I would like to track some module dependencies during the Node.js-based build.  The amodro-trace's `trace()` fits well, but it gathers all dependencies of the traced module as a flat array.  I would like to be able to check the static dependencies from the top `define()` statement too.

Let's have an example of tracing dependencies of the well-known Marionette.js library:

![image](https://cloud.githubusercontent.com/assets/733193/10869962/10e54066-80bf-11e5-9c98-c1ef48ea1588.png)

This is the output of the `trace()` result:

``` json
[
  {
    "id": "underscore", "path": "/projects/test/underscore.js"
  },
  {
    "id": "jquery", "path": "/projects/test/jquery.js"
  },
  {
    "id": "backbone", "path": "/projects/test/backbone.js"
  },
  {
    "id": "marionette", "path": "/projects/test/marionette.js"
  }
]
```

If I liked to see dependencies of the "backbone" module, I would have to run `trace()` for this module too and so on, downto the deepest dependencies.  This would be the desired output; _"dependencies"_ are IDs of the the immediate static module dependencies and _"parent"_ is the previous module, which caused loading of the current module bu including it in its static dependencies:

``` json
[
  {
    "id": "underscore", "path": "/projects/test/underscore.js",
    "dependencies": [],
    "parent": "marionette"
  },
  {
    "id": "jquery", "path": "/projects/test/jquery.js",
    "dependencies": [],
    "parent": "backbone"
  },
  {
    "id": "backbone", "path": "/projects/test/backbone.js",
    "dependencies": ["underscore", "jquery"],
    "parent": "marionette"
  },
  {
    "id": "marionette", "path": "/projects/test/marionette.js",
    "dependencies": ["underscore", "backbone"],
    "parent": null
  }
]
```

It would be possible to run `trace()` for every module possible, but it turned up too slow in our project.  Probably because every `trace()` uses a new loader.  The desired information was available during the first tracing, so why not including it in the produced output?

I thought that I could use `keepLoader: true` to access the `context.registry` when the `trace()` finished, but the registry was empty.  The information was still available when the `onResourceLoad` was called.  Is the registry used only temporarily during the dependency loading?

I tried to "back up" information from the registry, which I needed, in the `onResourceLoad` and add it to the `trace()` output in 6c4791e935f5979997e57454b8789ce3a25fd9a6.  Is this something you would like to see in your project?  If not, what would you recommend for this requirement?

Thank you!

**Comment by jrburke** [comment]
I just pushed a 0.0.8 release that includes the dependencies in the output. I did not do the parent part though, as I thought that could be computed after the fact given the other relations. The new output is shown in [this API section](https://github.com/amodrojs/amodro-trace#amodro-trace-1).

I was thinking of providing a separate module that would add  the "parent" data, maybe as an "amodro-dependents" adding a "dependents" field to each ID. Do you think that secondary processing step might work for that part?

**Comment by jrburke** [comment]
I just added a "dependents" field in #4, released as 0.0.9.
